pegasus/sites.v3/csedweek.org/views/map.haml

```diff
@@ -1,182 +1,32 @@
+%link{rel: "stylesheet", type: "text/css", href: "/css/map.css"}
+%link{href:'https://api.mapbox.com/mapbox-gl-js/v1.0.0/mapbox-gl.css', rel:'stylesheet'}
+%script{src:'https://api.mapbox.com/mapbox-gl-js/v1.0.0/mapbox-gl.js'}
+%script{src: webpack_asset_path("js/hourofcode.com/views/hoc_events_map.js"), defer:true}
+
 #hocmap
-  %link{rel: "stylesheet", type: "text/css", href: "/css/map.css"}
   %h2.mapheader
-    %a{:href=> "https://hourofcode.com/events"}
+    %a{href: CDO.hourofcode_url('/events')}
       = I18n.t(:csedweek_map_header).gsub('#', format_integer_with_commas(fetch_hoc_metrics['total_hoc_count']))
-    %a{:href=> "https://hourofcode.com/#join"}
+    %a{href: CDO.hourofcode_url('/#join')}
       %button.signup-button
         = I18n.t(:host_an_hour)
 
+  %div{id: 'mapbox-map'}
+
+  %div{id: 'belowmaplegend', class: 'legend'}
+    .legend-title= hoc_s(:map_legend_title)
+    .color.legend-hoc-event
+    %div= hoc_s(:map_legend_hoc_events)
+    .color.legend-special-event
+    %div= hoc_s(:map_legend_cs_tech_jam)
+
   .row
     .col-sm-12
-      #gmap.full-width
       .footnote
-        The Hour of Code map only shows the first 200,000 registered events. All registered events with a valid address can be viewed on
-        %a{href: resolve_url('/events')}
+        All registered events with a valid address can be viewed on
```

Comment:
can we add a follow-up work item to translate this string?

Reply:
https://codedotorg.atlassian.net/browse/FND-868